cpp/dolfinx/io/ADIOS2Writers.h

```diff
@@ -867,6 +867,330 @@ void vtx_write_mesh_from_space(adios2::IO& io, adios2::Engine& engine,
 }
 } // namespace impl_vtx
 
+/// @privatesection
+namespace impl_checkpoint
+{
+/// Write mesh to file using VTX format
```

Comment:
How is this different from the current VTX mesh writer?

Reply:
VTX mesh writer writes local partitions of the mesh (local indices) to separate state arrays in a .bp file.
This means that it cannot be read in on any other number of processes than the number that have written them to file (making them useless for checkpointing).

This mesh writer uses global mesh indices and writes everything to a global array.